File: examples/typescript/multi-instance.ts
/**
 * Example: Managing multiple OpenClaw Gateway instances
 */
import { InstanceManager } from '@minions-openclaw/core';

async function main() {
  const manager = new InstanceManager();

  const prod = await manager.register('production', 'ws://prod.example.com:8080', 'prod-token');
  const staging = await manager.register('staging', 'ws://staging.example.com:8080', 'staging-token');
  const local = await manager.register('local', 'ws://localhost:8080');

  console.log('Registered instances:');
  console.log('  Production:', prod.id);
  console.log('  Staging:', staging.id);
  console.log('  Local:', local.id);

  const instances = await manager.list();
  console.log('\nTotal instances:', instances.length);

  for (const inst of instances) {
    console.log(`  ${inst.title}: ${inst.fields['url']} [${inst.fields['status']}]`);
  }

  await manager.remove(prod.id);
  await manager.remove(staging.id);
  await manager.remove(local.id);
  console.log('\nCleaned up all instances');
}

main().catch(console.error);

File: examples/python/multi_instance.py
"""Example: Managing multiple OpenClaw Gateway instances."""
from minions_openclaw import InstanceManager
from pathlib import Path
import json

DATA_FILE = Path.home() / '.openclaw-manager' / 'data.json'


def main():
    # Clean up before example
    if DATA_FILE.exists():
        DATA_FILE.unlink()

    manager = InstanceManager()

    prod = manager.register('production', 'ws://prod.example.com:8080', 'prod-token')
    staging = manager.register('staging', 'ws://staging.example.com:8080', 'staging-token')
    local = manager.register('local', 'ws://localhost:8080')

    print('Registered instances:')
    print(f'  Production: {prod.id}')
    print(f'  Staging: {staging.id}')
    print(f'  Local: {local.id}')

    instances = manager.list()
    print(f'\nTotal instances: {len(instances)}')

    for inst in instances:
        print(f"  {inst.title}: {inst.fields['url']} [{inst.fields['status']}]")

    manager.remove(prod.id)
    manager.remove(staging.id)
    manager.remove(local.id)
    print('\nCleaned up all instances')


if __name__ == '__main__':
    main()
